model/src/main/kotlin/config/OrtConfiguration.kt

```diff
@@ -57,16 +57,22 @@ data class OrtConfiguration(
     val advisor: Map<String, AdvisorConfiguration>? = null
 ) {
     companion object {
+        /**
+         * The name of the embedded reference configuration resource.
+         */
+        const val REFERENCE_RESOURCE = "/reference.conf"
+
         /**
          * Load the [OrtConfiguration]. The different sources are used with this priority:
          *
          * 1. [Command line arguments][args]
          * 2. [Configuration file][file]
+         * 3. [Configuration resource][resource]
          *
          * The configuration file is optional and does not have to exist. However, if it exists, but does not
          * contain a valid configuration, an [IllegalArgumentException] is thrown.
          */
-        fun load(args: Map<String, String>? = null, file: File? = null): OrtConfiguration {
+        fun load(args: Map<String, String>? = null, file: File? = null, resource: String? = null): OrtConfiguration {
```

Comment:
Is there a concrete use case that a configuration is loaded from a resource except from test code? Otherwise, I would rather not introduce a new mechanism to load a configuration source as the file-based mechanism works well for tests.

My concern is that the role of the reference configuration can now be confused to represent the default configuration - which it is not; it is a sample configuration.

Reply:
> Is there a concrete use case that a configuration is loaded from a resource except from test code?

Well, yes, `ConfigCommand` needs it to be able to display it.

> My concern is that the role of the reference configuration can now be confused to represent the default configuration

Shouldn't the fact that `ConfigCommand` allows to display *both* the reference configuration and the default configuration make it rather clear that they are different things?

Reply:
The points that confused me were that a) the `load()` function was added a parameter for a resource and b) `reference.conf` was moved to resources. But in fact, this file is never loaded by production code via this mechanism, only by test classes. (`ConfigCommand` loads it directly from the classpath.)

So I am not sure whether this `resource` parameter in the `load()` function is actually needed as the test code can continue to load the file like before (just with an adapted path).

Reply:
You mean at the example of `NexusIqConfigurationTest` to change

    val referenceOrtConfig = OrtConfiguration.load(file = File("src/test/assets/reference.conf"))

into

    val referenceOrtConfig = OrtConfiguration.load(file = File("src/main/resources/reference.conf"))

Hmm, isn't it a bit dirty to access a resource directly from the file system, instead of actually loading it as a resource? I'm undecided. But I like the fact that this avoids changes to `OrtConfiguation`. @mnonnenmacher @fviernau what you you think, would this be dirty?

Reply:
`reference.conf` had been originally located under resources and was later moved to test assets. So the mechanisms to access it in tests should have been in place in the past.

Reply:
Which doesn't mean that mechanism was good back then 😉 But anyway, I've followed your proposal, as it also helps to keep the diff smaller.